File: sprint6/By languages/JS/lesson11code3.js
// Вызываем функцию поиска пути
const [dist, currentPoint] = escapeFromMaze(n, m, field, start_point);
const path = [];

// Строим путь от текущей точки до стартовой точки
while (!isEqual(currentPoint, start_point)) {
    path.push(currentPoint);
    currentPoint = prev[currentPoint];
}

// Условие выхода из цикла составлено так, что начальная точка не будет добавлена в путь, поправим это.
path.push(start_point);

// Разворачиваем путь, так как первой мы добавляли конечную точку маршрута.
path.reverse();

File: sprint6/By languages/kotlin/lesson11code3.kt
// Вызываем функцию поиска пути
val (dist: Int, currentPoint: Pair<Int, Int>) = escapeFromMaze(n, m, field, start_point)
val path = mutableListOf<Pair<Int, Int>>()

// Строим путь от текущей точки до стартовой точки
while (currentPoint != start_point) {
    path.add(currentPoint)
    currentPoint = prev[currentPoint]!!
}

// Условие выхода из цикла составлено так, что начальная точка не будет добавлена в путь, поправим это.
path.add(start_point)

// Разворачиваем путь, так как первой мы добавляли конечную точку маршрута.
path.reverse()
